openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -205,3 +212,80 @@ def award_program_certificates(self, username):
         LOGGER.info('User %s is not eligible for any new program certificates', username)
 
     LOGGER.info('Successfully completed the task award_program_certificates for username %s', username)
+
+
+def post_course_certificate(client, username, certificate):
+    """
+    POST a certificate that has been updated to Credentials
+    """
+    client.credentials.post({
+        'username': username,
+        'status': 'awarded' if certificate.is_valid() else 'revoked',  # Only need the two options at this time
+        'credential': {
+            'course_run_key': str(certificate.course_id),
+            'mode': certificate.mode,
+            'type': COURSE_CERTIFICATE,
+        }
+    })
+
+
+@task(bind=True, ignore_result=True, routing_key=ROUTING_KEY)
+def award_course_certificates(self, username, course_run_key):
+    LOGGER.info('Running task award_course_certificates for username %s', username)
+
+    countdown = 2 ** self.request.retries
+
+    # If the credentials config model is disabled for this
+    # feature, it may indicate a condition where processing of such tasks
+    # has been temporarily disabled.  Since this is a recoverable situation,
+    # mark this task for retry instead of failing it altogether.
+
+    if not CredentialsApiConfig.current().is_learner_issuance_enabled:
+        LOGGER.warning(
+            'Task award_course_certificates cannot be executed when credentials issuance is disabled in API config',
+        )
+        raise self.retry(countdown=countdown, max_retries=MAX_RETRIES)
+
+    try:
+        try:
+            user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            LOGGER.exception('Task award_course_certificates was called with invalid username %s', username)
+            # Don't retry for this case - just conclude the task.
+            return
+        # Get the cert for the course key and username if it's both passing and available in professional/verified
+        try:
+            certificate = GeneratedCertificate.eligible_certificates.get(
+                user=user.id,
+                course_id=course_run_key
+            )
+        except GeneratedCertificate.DoesNotExist:
+            LOGGER.exception(
+                'Task award_course_certificates was called without Certificate found for %s to user %s',
+                course_run_key,
+                username
+            )
+            return
+        if certificate.mode in GeneratedCertificate.VERIFIED_CERTS_MODES:
+            try:
+                course_overview = CourseOverview.get_from_id(course_run_key)
+            except (CourseOverview.DoesNotExist, IOError):
+                LOGGER.exception(
+                    'Task award_course_certificates was called without course overview data for course %s',
+                    course_run_key
+                )
+                return
+            if certificates_viewable_for_course(course_overview):
```

Comment:
What happens when a course cert is awarded, but not visible. And then later, it becomes visible because the deadline passes or whatever. How would we notice and kick off this job again?